hw/ip/aon_timer/doc/_index.md

```diff
@@ -0,0 +1,102 @@
+The wakeup timer can be used like a real-time clock for long periods of low-power mode (though it does not give any guarantees of time-accuracy TODO: specify accuracy).
+
+### AON Watchdog timer
+
+The always-on watchdog timer behaves similarly to the wakeup timer.
+It has an independent count starting at 0 which slowly ticks upwards.
+When the first threshold is met or exceeded, a level wakeup signal (if enabled) is generated to the power manager.
+Simultaneously, a level IRQ signal is also generated to the processor.
+
+If the system is in a low power state, the wakeup signal asks the power manager to wake the system such that the IRQ can be serviced.
+If the system is not in low power mode, the IRQ is immediately serviced.
+Both the wakeup and the IRQ signals remain asserted until system reset or explicit acknowledgement by software.
+This first threshold is known as the watchdog bark.
+
+When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which will reset the whole system.
+This is independent of the IRQ sent as part of the watchdog bark.
+The system reset also resets the always-on timer, so software is not required to directly acknowledge anything after a watchdog reset.
+
+To prevent the watchdog bark or bite, software is expected to periodically reset the count when operating normally.
+This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
+
+Since this timer functions as a watchdog, it has two additional functions not present in the always-on wakeup timer:
+* Watchdog configuration lock
+* Watchdog pause in sleep
+
+Unlike the wakeup timer, the watchdog timer configuration might need to be locked by firmware until the next system reset.
+This allows the option of preventing firmware from accidentally or maliciously disabling the watchdog.
+
+The "pause in sleep" option controls whether the watchdog timer continues to count in low-power mode.
+This allows configurations where the watchdog timer can remain programmed and locked while the device is put to sleep for relatively long periods, controlled by the wakeup timer.
+Without this feature, the watchdog timer might wake up the core prematurely by triggering a watchdog bark.
+
+# Theory of Operations
+
+## Block Diagram
+
+![AON Timer Block Diagram](aon_timer_block_diagram.svg)
+
+See the block diagram for high level function and connectivity.
+The timer interacts with the CPU core and the power manager and reset manager to drive wakeup / reset events and interrupts.
+There is also an extra input to tell the counter whether to run ("counter-run").
+This is used to stop the watchdog timer running when in debugging mode or when the alert handler has put the system in a "killed" state.
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/aon_timer/data/aon_timer.hjson" >}}
+
+## Design Details
+
+The always-on timer will run on a ~200KHz clock.
```

Comment:
@rswarbrick 
do you guys think it makes sense for this kind of information to link to the [ast documentation](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/ip/ast/doc/_index.md)?

Reply:
Hmm, maybe. It depends how self-contained and modular we want the blocks to be. All the comportability stuff seems to suggest we want to be able to grab IP blocks and slap them down in other systems. If we're doing that, then "upward" links like that in the documentation are probably confusing. Of course, in reality, we'll probably never do it (and will invent completely new tooling for the next major project, because one always does...), in which case we should totally link everything together. So... not sure!

Reply:
It would be useful (for software) to specify things like the clock precision to know what the error bars look like on timeouts. I don't think any of that kind of information is (or ever will be) in the AST documentation though.